examples/basic/utils_level_zero.h

```diff
@@ -0,0 +1,236 @@
+/*
```

Comment:
why is this file (and gpu_shared_memory.c) in `basic` dir? I find it a little confusing, tbh

Reply:
what do you suggest? All examples will be installed in the /examples folder (without any subfolders) anyway.

Reply:
not done

Reply:
@lukaszstolarczuk I think we could do this in separate PR among with changes that adds CMake files